integration/automation-extensions/src/main/java/org/wso2/esb/integration/common/extensions/carbonserver/CarbonServerManager.java

```diff
@@ -144,7 +149,12 @@ public synchronized void startServerUsingCarbonHome(String carbonHome, Map<Strin
                 }
             }));
 
-            managementPort = 9154 + portOffset;
+            if (StringUtils.isNotEmpty(commandMap.get("managementPort"))) {
```

Comment:
better to retrieve both from system prop . then no need to edit automation xml to run tests with EI ..

Reply:
From where we will set the System.property in connectors? (Classes inside connector and integration base gets hit after server starts)

Reply:
then how is that handled in scriptName = System.getProperty("startupScript");

Reply:
it's resolving to null and reading the value from scriptName = commandMap.get("startupScript");. I can remove the scriptName = System.getProperty("startupScript") part.